src/main/java/com/ontotext/kafka/producer/GraphDBProducer.java

```diff
@@ -28,30 +33,30 @@ public void publish() {
         for (String file : this.allFiles) {
             try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
                 String line = reader.readLine();
-                String key = null;
+                byte[] key = null;
                 if (this.keySeparator != null && line.contains(this.keySeparator)) {
-                    key = line.split(this.keySeparator)[0];
+                    key = line.split(this.keySeparator)[0].getBytes();
```

Comment:
No, do not add the key on each line. Add a second file from which keys can be read as well as a property for the number of files per key or number of lines per key.